ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,106 @@
+from typing import Callable, Optional, Union
+
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.gan.utils import InceptionModel, _BaseInceptionMetric
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import reinit__is_reduced, sync_all_reduce
+
+__all__ = ["InceptionScore"]
+
+
+class InceptionScore(_BaseInceptionMetric):
+    r"""Calculates Inception Score.
+
+    .. math::
+       \text{IS(G)} = \exp(\frac{1}{N}\sum_{i=1}^{N} D_{KL} (p(y|x^{(i)} \parallel \hat{p}(y))))
+
+    where :math:`p(y|x)` is the conditional probability of image being the given object and
+    :math:`p(y)` is the marginal probability that the given image is real, `G` refers to the
+    generated image and :math:`D_{KL}` refers to KL Divergence of the above mentioned probabilities.
+
+    More details can be found in `Barratt et al. 2018`__.
+
+    __ https://arxiv.org/pdf/1801.01973.pdf
+
+
+    Args:
+        num_features: number of features predicted by the model or number of classes of the model. Default
+            value is 1000.
+        feature_extractor: a torch Module for predicting the probabilities from the input data.
+            If neither ``num_features`` nor ``feature_extractor`` are defined, by default we use an ImageNet
+            pretrained Inception Model. Please note that the class object will be implicitly converted to device
+            mentioned in the ``device`` argument.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan.IS import InceptionScore
+            import torch
+
+            images = torch.rand(10, 3, 299, 299)
+
+            m = InceptionScore()
+            m.update(images)
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_features: Optional[int] = None,
+        feature_extractor: Optional[torch.nn.Module] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+
+        self._default_channels = 1000
+        self._default_eval_model = InceptionModel
+        self._default_args = False
```

Comment:
Set those as input args for `self._check_input` directly. Let's try to remove those instance args